app/services/pupil_premium_importer.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require "csv"
+
+class PupilPremiumImporter
+  attr_reader :logger
+  attr_reader :start_year
+  attr_reader :source_file
+
+  def initialize(logger, start_year = Time.zone.now.year, source_file = nil)
+    @logger = logger
+    @start_year = start_year
+    @source_file = source_file
+  end
+
+  def run
+    CSV.foreach(data_file, headers: true, encoding: "ISO-8859-1:UTF-8") do |row|
+      update_school_premium(row)
+    end
+  end
+
+private
+
+  def data_file
+    source_file || Rails.root.join("data/pupil_premium.csv")
```

Comment:
Rails.root.join("data", "pupil_premium.csv") would be platform independent. I also wonder if we could just do this in the initialiser i.e. @source_file = source_file || Rails.root.join("data", "pupil_premium.csv")

Reply:
I think that will lead to 
```
RuboCop: Please use `Rails.root.join('path/to')` instead. [Rails/FilePath]
```

Reply:
Yep, exactly